src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,143 @@
+        if (prev == null) {
+            this.head = add;
+        } else {
+            prev.next = add;
+        }
+        willNext.prev = add;
+        size++;
     }
 
     @Override
-    public boolean addAll(List<T> list) {
-        return false;
+    public boolean addAll(List<T> items) {
+        for (T item : items) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T set, int index) {
+        Node<T> request = getNode(index);
+        T item = request.item;
+        request.item = set;
+        return item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> request = getNode(index);
+        if (request.next == null) {
+            this.tail = request.prev;
+        } else if (request.prev == null) {
+            this.head = request.next;
+        } else {
+            request.prev.next = request.next;
+            request.next.prev = request.prev;
+        }
+        size--;
+        return request.item;
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(T item) {
+        for (int i = 0; i < this.size; i++) {
+            Node<T> request = getNode(i);
+            if (request.item == item || request.item.equals(item) && request.item != null) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return this.size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item, Node<T> prev, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
     }
+
+    private Node<T> getNode(int index) {
+        indexException(index);
+        if (index <= this.size / 2) {
+            return searchFromHead(index);
+        } else {
+            return searchFromTail(index);
+        }
+    }
+
+    private Node<T> searchFromHead(int index) {
+        Node<T> request = this.head;
+        for (int i = 0; i < index; i++) {
+            request = request.next;
+        }
+        return request;
+    }
+
+    private Node<T> searchFromTail(int index) {
+        Node<T> request = this.tail;
+        for (int i = this.size - 1; i > index; i--) {
+            request = request.prev;
+        }
+        return request;
+    }
+
+    private void indexException(int index) {
+        if (index >= this.size || index < 0) {
+            throw new IndexOutOfBoundsException("Sir, your index is out of bounds");
```

Comment:
```suggestion
            throw new IndexOutOfBoundsException("Sir, your index is out of bounds");
```
what if your code will be used by madam? - better use classic message and take it to constant 
private static final String OUT_OF_BOUNDS_MESSAGE = "..."